sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -308,31 +335,69 @@ public static BinaryData fromObject(Object data, ObjectSerializer serializer) {
     }
 
     /**
-     * Returns a byte array representation of this {@link BinaryData}.
+     * Creates a {@link BinaryData} that uses {@link Path} as its data. This method checks for the existence of
+     * the file at the time of creating an instance of {@link BinaryData}. The file, however, is not read until there
+     * is an attempt to read the contents of the returned BinaryData instance.
+     *
+     * <p><strong>Create an instance from a file</strong></p>
+     *
+     * {@codesnippet com.azure.core.util.BinaryData.fromFile}
+     *
+     * @param file The {@link Path} that will be the {@link BinaryData} data.
+     * @return A new {@link BinaryData}.
+     * @throws NullPointerException If {@code file} is null.
+     */
+    public static BinaryData fromFile(Path file) {
+        return fromFile(file, STREAM_READ_SIZE);
+    }
+
+    /**
+     * Creates a {@link BinaryData} that uses {@link Path file} as its data. This method checks for the existence of
+     * the file at the time of creating an instance of {@link BinaryData}. The file, however, is not read until there
+     * is an attempt to read the contents of the returned BinaryData instance.
+     *
+     * <p><strong>Create an instance from a file</strong></p>
+     *
+     * {@codesnippet com.azure.core.util.BinaryData.fromFile#Path-int}
+     *
+     * @param file The {@link Path} that will be the {@link BinaryData} data.
+     * @param chunkSize The requested size for each read of the path.
+     * @return A new {@link BinaryData}.
+     * @throws NullPointerException If {@code file} is null.
+     * @throws IllegalArgumentException If {@code offset} or {@code length} are negative or {@code offset} plus {@code
+     * length} is greater than the file size or {@code chunkSize} is less than or equal to 0.
+     * @throws UncheckedIOException if the file does not exist.
+     */
+    public static BinaryData fromFile(Path file, int chunkSize) {
+        return new BinaryData(new FileContent(file, chunkSize));
+    }
+
+    /**
+     * Returns a byte array representation of this {@link BinaryData}. This method returns a reference to the
+     * underlying byte array. Modifying the contents of the returned byte array will also change the content of this
+     * BinaryData instance. To safely update the byte array, it is recommended to make a copy of the contents first.
```

Comment:
Might pay to note that if the BinaryData represents a file stream, etc, that it will not modify the source data.